File: 696_Count_Binary_Substrings/solution.c
int countBinarySubstrings(char* s) {
    int result = 0, prev = 0, curr = 1;
    for (int i = 1; i < strlen(s); i++) {
        if (s[i] == s[i - 1]) curr++;
        else { prev = curr; curr = 1; }
        if (curr <= prev) result++;
    }   return result;
}

File: 696_Count_Binary_Substrings/solution.py
class Solution:
    def countBinarySubstrings(self, s: str) -> int:
        result, prev, curr = 0, 0, 1
        for i in range(1, len(s)):
            if s[i] == s[i - 1]: curr += 1
            else: 
                prev = curr
                curr = 1
            if curr <= prev: result += 1
        return result